src/verilog.y

```diff
@@ -1558,9 +1563,34 @@ port<nodep>:                    // ==IEEE: port
                           VARDTYPE(dtp); VARIOANSI();
                           addNextNull($$, VARDONEP($$, $6, $7)); }
         |       portDirNetE yINTERFACE                           portSig rangeListE sigAttrListE
-                        { $$ = nullptr; BBUNSUP($<fl>2, "Unsupported: generic interfaces"); }
+                        { $$ = $3;
+                          GRAMMARP->m_genericIfaceModule = true;
+                          VARDECL(GPARAM); VARIO(NONE);
+                          VARDTYPE(new AstParseTypeDType($<fl>2, VFwdType::GENERIC_INTERFACE));
+                          std::string uniqueName = "__VGIfaceParam" + $$->name();
+                          addNextNull($$, VARDONEA($$->fileline(), uniqueName, $4, $5));
+                          VARDECL(IFACEREF);
+                          GRAMMARP->m_pinNum = PINNUMINC();
+                          AstIfaceGenericDType* const refdtypep = new AstIfaceGenericDType($<fl>2);
+                          VARDTYPE(refdtypep); VARIOANSI();
+                          addNextNull($$, VARDONEP($$, $4, $5));
+                          VARDECL(VAR);
+                        }
         |       portDirNetE yINTERFACE      '.' idAny/*modport*/ portSig rangeListE sigAttrListE
-                        { $$ = nullptr; BBUNSUP($<fl>2, "Unsupported: generic interfaces"); }
+                        {
+                          $$ = $5;
+                          GRAMMARP->m_genericIfaceModule = true;
+                          VARDECL(GPARAM); VARIO(NONE);
+                          VARDTYPE(new AstParseTypeDType($<fl>2, VFwdType::GENERIC_INTERFACE));
+                          std::string uniqueName = "__VGIfaceParam" + $$->name();
+                          addNextNull($$, VARDONEA($$->fileline(), uniqueName, $6, $7));
+                          VARDECL(IFACEREF);
+                          GRAMMARP->m_pinNum = PINNUMINC();
+                          AstIfaceGenericDType* const refdtypep = new AstIfaceGenericDType($<fl>2, $<fl>4, *$4);
+                          VARDTYPE(refdtypep); VARIOANSI();
+                          addNextNull($$, VARDONEP($$, $6, $7));
+                          VARDECL(VAR);
```

Comment:
Consider if should move most of this to a createGenericIface method at top of this file, and call here, to avoid duplicating code above.  (One argument would be modport of either nullptr or the modport)